Actions/EventAction.cpp

```diff
@@ -43,6 +43,8 @@ void EventAction::BeginOfEventAction(const G4Event*)
     fScinCollID = SDman->GetCollectionID(colNam = "detectorCollection");
   }
   fHistoManager->Clear();
+  fHistoManager->SetEventNumber(fEventID);
```

Comment:
Instead of using another iterator `fEventID` you can use the one that is already in Geant4, that does that, `event->GetEventID()`. Would need to add also variable in the method declaration `BeginOfEventAction(const G4Event* event)`

Reply:
I resolved it by removing zeroing of the fEvtIndex whenever clear is called for it, as it was cleared at the beginning of every event and the information of the event number was dropped for eventPack every new event. 
Now setting the evtIndex is done at the end of the event as: currentEvtIndex + 1
it works ok, but maybe we could think about small reorganisation of the code - connected to yours first comment about moving the selection of the decayChannel. But I do not know if we can do it now or maybe after some pre-/normal relase of the nwe version